ml3d/datasets/matterport_objects.py

```diff
@@ -0,0 +1,252 @@
+            [mat, np.array([[0., 0., 1., 0.]], dtype=mat.dtype)], axis=0)
+        return mat
+
+    def get_split(self, split):
+        """Returns a dataset split.
+
+        Args:
+            split: A string identifying the dataset split that is usually one of
+            'training', 'test', 'validation', or 'all'.
+
+        Returns:
+            A dataset split object providing the requested subset of the data.
+        """
+        return MatterportObjectsSplit(self, split=split)
+
+    def get_split_list(self, split):
+        """Returns the list of data splits available.
+
+        Args:
+            split: A string identifying the dataset split that is usually one of
+            'training', 'test', 'validation', or 'all'.
+
+        Returns:
+            A dataset split object providing the requested subset of the data.
+
+        Raises:
+            ValueError: Indicates that the split name passed is incorrect. The
+            split name should be one of 'training', 'test', 'validation', or
+            'all'.
+        """
+        if split in ['train', 'training']:
+            return self.train_files
+        elif split in ['test', 'testing']:
+            return self.test_files
+        elif split in ['val', 'validation']:
+            return self.val_files
+        elif split in ['all']:
+            return self.train_files + self.val_files + self.test_files
+        else:
+            raise ValueError("Invalid split {}".format(split))
+
+    def is_tested(self, attr):
+        """Checks if a datum in the dataset has been tested.
+
+        Args:
+            dataset: The current dataset to which the datum belongs to.
+            attr: The attribute that needs to be checked.
+
+        Returns:
+            If the dataum attribute is tested, then resturn the path where the
+            attribute is stored; else, returns false.
+        """
+        pass
+
+    def save_test_result(self, results, attrs):
+        """Saves the output of a model.
+
+        Args:
+            results: The output of a model for the datum associated with the
+            attribute passed.
+            attrs: The attributes that correspond to the outputs passed in
+            results.
+        """
+        pass
+
+
+class MatterportObjectsSplit():
+
+    def __init__(self, dataset, split='train'):
+        self.cfg = dataset.cfg
+        path_list = dataset.get_split_list(split)
+        log.info("Found {} pointclouds for {}".format(len(path_list), split))
+
+        self.path_list = path_list
+        self.split = split
+        self.dataset = dataset
+
+    def __len__(self):
+        return len(self.path_list)
+
+    def get_data(self, idx):
+        pc_path = self.path_list[idx]
+        label_path = pc_path.replace('pc', 'boxes').replace('.bin', '.txt')
+
+        pc = self.dataset.read_lidar(pc_path)
+        label = self.dataset.read_label(label_path)
+
+        data = {
+            'point': pc,
+            'full_point': pc,
```

Comment:
Do we need this `full_point` attribute ?. We have this in KITTI as it only annotates objects which are visible in front camera, so we pass only the cropped region in `point` attribute.

Reply:
No, not needed.

I deleted both `'full_point'` and `'feat'` as neither are used. I left `'calib'` because it is expected to exist for many of the operations in `ml3d/datasets/utils/transforms.py`